src/module.c

```diff
@@ -1045,6 +1046,47 @@ void swapDbEventCallback(RedisModuleCtx *ctx, RedisModuleEvent e, uint64_t sub,
     }
 }
 
+void ShardingEvent(RedisModuleCtx *ctx, RedisModuleEvent eid, uint64_t subevent, void *data) {
+    /**
+     * On sharding event we need to do couple of things depends on the subevent given:
+     *
+     * 1. REDISMODULE_SUBEVENT_SHARDING_SLOT_RANGE_CHANGED
+     *    On this event we know that the slot range changed and we might have data
+     *    which are no longer belong to this shard, we must ignore it on searches
+     *
+     * 2. REDISMODULE_SUBEVENT_SHARDING_TRIMMING_STARTED
+     *    This event tells us that the trimming process has started and keys will start to be
+     *    deleted, we do not need to do anything on this event
+     *
+     * 3. REDISMODULE_SUBEVENT_SHARDING_TRIMMING_ENDED
+     *    This event tells us that the trimming process has finished, we are not longer
+     *    have data that are not belong to us and its safe to stop checking this on searches.
+     */
+    if (eid.id != REDISMODULE_EVENT_SHARDING) {
+        RedisModule_Log(rts_staticCtx, "warning", "Bad event given, ignored.");
+        return;
+    }
+
+    switch (subevent) {
+        case REDISMODULE_SUBEVENT_SHARDING_SLOT_RANGE_CHANGED:
```

Comment:
@OfirMos I think this is used for OSS clustering, which doesn't have trimming support

Reply:
@danni-m this subevent doesn't appear in redismodule.h or in the module_api docs so it's RS only.
This impl was being cloned from Redis Search so the correctness is transitive I guess :)